src/resource/nonRdfData.ts

```diff
@@ -243,6 +243,36 @@ export async function overwriteFile(
   return Object.assign(blobClone, { internal_resourceInfo: resourceInfo });
 }
 
+/**
+ * @hidden
+ * This function feels unnecessarily complicated, but is required in order to
+ * have Headers according to type definitions in both Node and browser environments.
+ * This might require a fix upstream to be cleaned up.
+ *
+ * @param headersToFlatten A structure containing headers potentially in several formats
+ */
+export function flattenHeaders(
+  headersToFlatten: Headers | Record<string, string> | string[][] | undefined
+): Record<string, string> {
+  if (typeof headersToFlatten === "undefined") {
+    return {};
+  }
+  const flatHeaders: Record<string, string> = {};
+
+  // If the headers are already a Record<string, string>,
+  // they can directly be returned.
+  if (typeof headersToFlatten.forEach !== "function") {
+    // FIXME: This will break when passed a string[][]
+    //        (as shown when the type assertions are removed).
+    return headersToFlatten as Record<string, string>;
+  }
+
+  (headersToFlatten as Headers).forEach((value: string, key: string) => {
+    flatHeaders[key] = value;
+  });
+  return flatHeaders;
```

Comment:
The damn `string[][]` form isn't even documented anywhere. Also, I think it's technically an `Array<[string, string]>`, i.e. every element has two values.

Anyway, maybe it's best to deal with this right now as well? (And maybe also avoid `forEach`, since [MDN](https://developer.mozilla.org/en-US/docs/Web/API/Headers) doesn't have a page for it and doesn't even list it in its compatibility table.)

```suggestion
  // If the headers are already a Record<string, string>,
  // they can directly be returned.
  if (typeof headersToFlatten.entries !== "function" && !Array.isArray(headersToFlatten)) {
    return headersToFlatten;
  }

  const flatHeaders: Record<string, string> = {};
  const headerKeys: Array<[string, string]> = (typeof headersToFlatten.entries !== "function")
    ? headersToFlatten
    : Array.from(headersToFlatten.entries());
    
  headerKeys.forEach(([key, value]) => {
    flatHeaders[key] = value;
  });
  return flatHeaders;
```

Reply:
I had to make a few twists on your suggestion, but the idea is in f7363578f85359d64cb48d726fd1847b9c323d1f